dora/core/common/src/main/java/alluxio/membership/DefaultServiceEntity.java

```diff
@@ -75,23 +78,55 @@ public CloseableClient getKeepAliveClient() {
   }
 
   /**
-   * Serialize the ServiceEntity to output stream.
-   * @param dos
-   * @throws IOException
+   * @return lease
+   */
+  public AlluxioEtcdClient.Lease getLease() {
+    return mLease;
+  }
+
+  /**
+   * Set lease.
+   * @param lease
+   */
+  @GuardedBy("mLock")
+  public void setLease(AlluxioEtcdClient.Lease lease) {
+    mLease = lease;
+  }
+
+  /**
+   * Get the revision number of currently registered DefaultServiceEntity
+   * on ETCD.
+   * @return revision number
+   */
+  public long getRevisionNumber() {
+    return mRevision;
+  }
+
+  /**
+   * Set revision number.
+   * @param revisionNumber
+   */
+  public void setRevisionNumber(long revisionNumber) {
+    mRevision = revisionNumber;
+  }
+
+  /**
+   * @return lock for atomically modifying certain fields
    */
-  public void serialize(DataOutputStream dos) throws IOException {
-    dos.writeUTF(mServiceEntityName);
-    dos.writeLong(mRevision);
+  ReentrantLock getLock() {
+    return mLock;
   }
 
   /**
-   * Deserialize the ServiceEntity from input stream.
-   * @param dis
-   * @throws IOException
+   * Convert a DefaultServiceEntity into a json string.
+   * @param entity
+   * @return json string
    */
-  public void deserialize(DataInputStream dis) throws IOException {
-    mServiceEntityName = dis.readUTF();
-    mRevision = dis.readLong();
+  public static String toJson(DefaultServiceEntity entity) {
```

Comment:
maybe leave the implement to decide the serialization type? license uses the service entity and it's no need to use json for now. maybe we can provide a interface like this?
```java
public static byte[] serialize(DefaultServiceEntity entity) throws IOException {...}
public static DefaultServiceEntity deserialize(byte[] buf) throws IOException {...}
```